client/src/battle/states/card_select_state.rs

```diff
@@ -188,13 +250,36 @@ impl State for CardSelectState {
             return;
         };
 
+        // let globals = game_io.resource::<Globals>().unwrap();
+        let mut pending_sfx = Vec::new();
+
         let globals = game_io.resource::<Globals>().unwrap();
         let selection = &self.player_selections[player.index];
         let selected_item = resolve_selected_item(player, selection);
 
         // update frame
         if let SelectedItem::Card(i) = selected_item {
             self.ui.update_card_frame(game_io, player, i);
+
+            let card = &player.deck[i];
+            let globals = game_io.resource::<Globals>().unwrap();
+
+            let card_packages = &globals.card_packages;
+            let namespace = player.namespace();
+
+            let is_dark = card_packages
+                .package_or_override(namespace, &card.package_id)
+                .is_some_and(|package| package.card_properties.card_class == CardClass::Dark);
+
+            if is_dark {
+                let fade_sprite = &mut resources.fade_sprite.clone();
+                fade_sprite.set_bounds(Rect::from_corners(Vec2::ZERO, RESOLUTION_F));
+                fade_sprite.set_color(resources.ui_fade_color.clone());
+                sprite_queue.draw_sprite(&fade_sprite);
```

Comment:
I was thinking ui_fade_color should be modified in `update()` / a function called by `update()`, while drawing the fade_sprite could be handled in battle_scene.rs's `draw()`, and reset in battle_simulation.rs's `pre_update()`

I'm starting to wonder if we should just store this fade_sprite on CardSelectState.